### PR TITLE
Extended infos are working again (posters, releases, crew and cast)

### PR DESCRIPTION
**Pull request by Flink**
Ok, lots of work here :) All tests are green and I’ve added one for the new syntax of Tmdb::api_call.

I’ve modified Tmdb::api_call so that it accepts an extra action in the first parameter. For example 'movie/images' to get 'movie/:id/images'. It also works with actions that do not require an id. ('search' for example)

So with this modification made, I added the retrieving of posters, cast, crew and releases when expand_results is true. Also, I create extra keys on posters and backdrops collections. They’re grouped in a 'sizes' collection to easily retrieve the wanted image url. I replaced the generated singletons with those keys since it wasn’t playing well when trying to marshal them.

And I also added configuration info (from the API configuration action) but it is static at the moment.

If you have questions or remarks, feel free to ask/tell me :)

### CODE REVIEW

File: test/setup/setup_api_key.rb [hunk 1]
```diff
@@ -2,9 +2,9 @@
   file_path = File.absolute_path('tmdb_api_key.txt', File.dirname(__FILE__))
 
   File.open(file_path) do |file|
-    Tmdb.api_key = file.read
+    Tmdb.api_key = file.read.chomp!
```
Comment (Irio) [inline test/setup/setup_api_key.rb]: What about `Tmdb.api_key = file.read.chomp`? `chomp!` returns nil if nothing was removed. It was causing some fails here.

File: lib/ruby-tmdb3/tmdb.rb [hunk 2]
```diff
@@ -100,23 +109,23 @@ def self.get_url(uri_str, limit = 10)
...
-      if(object.respond_to?(image_array_name))
-        image_array = object.send(image_array_name)
-        image_array.each_index do |x|
-          image_array[x] = image_array[x].image
-          image_array[x].instance_eval <<-EOD
-            def self.data
-              return Tmdb.get_url(self.url).body
-            end
-          EOD
+    ["posters", "backdrops"].each do |image_array_name|
```
Comment (Irio) [inline lib/ruby-tmdb3/tmdb.rb:114]: Could you add a comment about this block? Specially in the 118..125 lines; or maybe just add some blank lines to make them more easy to read.

**Comment by Irio** [comment]
Hi!

Great improvements, especially with Tmdb.api_call construct query and images feature.

I just made inline comments about some things that could be changed before merge. Thanks.

File: lib/ruby-tmdb3/tmdb.rb [hunk 3]
```diff
@@ -100,23 +109,23 @@ def self.get_url(uri_str, limit = 10)
...
+        urls = CONFIGURATION.images.send("#{image_array_name}_sizes").inject({}) do |hash, size|
+          hash[size] = {'url' => [CONFIGURATION.images.base_url, size, image.file_path].join}
+          hash
         end
+        image.sizes = DeepOpenStruct.load urls
       end
-      if(object.profile)
-        object.profiles = object.profile
-      end
+      #if(object.profile)
```
Comment (Irio) [inline lib/ruby-tmdb3/tmdb.rb]: Maybe remove this 3 commented lines?

**Comment by Flink** [comment]
Ok, thanks for your feedback :) I’ll do some updates.

File: test/setup/setup_api_key.rb (hunk 1 above)
Comment (Flink) [inline test/setup/setup_api_key.rb]: Yep sorry, I’ll fix this!

File: lib/ruby-tmdb3/tmdb.rb (hunk 2 above)
Comment (Flink) [inline lib/ruby-tmdb3/tmdb.rb:114]: Ok, I’ll try to add some comments, but I think it’s pretty self-explanatory :)

**Comment by Flink** [comment]
It should be ok now :)

**Pull request merged**